sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/models/SqlQuerySpecWithEncryption.java

```diff
@@ -84,9 +93,21 @@ Mono<Void> addEncryptionParameterAsync(String path, SqlParameter sqlParameter,
                                 byte[] cipherTextWithTypeMarker = new byte[cipherText.length + 1];
                                 cipherTextWithTypeMarker[0] = (byte) typeMarkerPair.getLeft().getValue();
                                 System.arraycopy(cipherText, 0, cipherTextWithTypeMarker, 1, cipherText.length);
-                                SqlParameter encryptedParameter = new SqlParameter(sqlParameter.getName(),
-                                    cipherTextWithTypeMarker);
+
+                                SqlParameter encryptedParameter;
+                                if (propertyName.equals(Constants.PROPERTY_NAME_ID)) {
+                                    // case: id does not support '/','\','?','#'. Convert Base64 string to Uri safe string
+                                    String base64UriSafeString =  convertToBase64UriSafeString(cipherTextWithTypeMarker);
+                                    System.out.println("encrypted id " + base64UriSafeString);
```

Comment:
remove system.out.println or replace with proper logger

Reply:
Isn't .Net also applying base64 encoding (with the additional hoops) for PK?

Reply:
Removed the println statement. Base 64 encoding is being applied only for the id field in .Net as well.